app/lib/actions/helpers/smart_proxy_sync_history_helper.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Helpers
+    module SmartProxySyncHistoryHelper
+      def self.included(base)
+        base.middleware.use ::Actions::Middleware::RecordSmartProxySyncHistory
+      end
+
+      def done?
+        is_done = super
+        if is_done
+          ::Katello::SmartProxySyncHistory.where(:id => output[:smart_proxy_history_ids]).update_all(finished_at: Time.now)
```

Comment:
hah, i just realized we're not actually checking for errors here.  We want to delete it if there are errors

Reply:
or just not update it as we discussed